frontend/app/routes/settings.tsx

```
@@ -1,12 +1,486 @@
+                className="px-4 py-2 bg-gray-800 border border-gray-700 rounded-lg hover:bg-gray-700 hover:border-orange-500 transition-all text-sm font-medium disabled:opacity-50"
+              >
+                {testingTor ? "Testing..." : "🔌 Test Tor"}
+              </button>
+            </div>
+
+            <div className="grid grid-cols-1 sm:grid-cols-3 gap-4">
+              <div>
+                <label className="block text-sm text-gray-400 mb-2">
+                  Control Port
+                </label>
+                <input
```

Biome (2.3.14)

[error] A form label must be associated with an input.

Consider adding a `for` or `htmlFor` attribute to the label element or moving the input element to inside the label element.

(lint/a11y/noLabelWithoutControl)